streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -442,6 +442,10 @@ public final void addSource(final Topology.AutoOffsetReset offsetReset,
                                      final String... predecessorNames) {
         Objects.requireNonNull(name, "name must not be null");
         Objects.requireNonNull(topic, "topic must not be null");
+        if (predecessorNames.length == 0) {
```

Comment:
No we need to check for `null` here, too?

Reply:
True, I think null can be explicitly passed as a parameter which would give `null` for `predecessorNames` rather than array containing `null`. I'll add a check.